scripts/envDFL.sh

```diff
@@ -0,0 +1,243 @@
+    fi
+
+    if eval "[ \"\$$name\" ]"; then
+      eval "value=\${$name}"
+      ##echo "Environment variable for $name already set : $value"
+      return
+    fi
+
+    if [ "$value" = UNDEFINED ]; then
+      error_exit "You must define the value of $name"
+    fi
+    export $name="$value"
+}
+
+ld_library_path_remove() {
+    export LD_LIBRARY_PATH=`echo -n $LD_LIBRARY_PATH | awk -v RS=: -v ORS=: '$0 != "'$1'"' | sed 's/:$//'`;
+}
+
+define_options() {
+    export_var_env DYNAFLOW_LAUNCHER_USAGE="Usage: `basename $0` [option] -- program to deal with DynaFlow Launcher
+
+where [option] can be:
+        =========== Build
+        build-user                              build DynaFlow Launcher
+        build-tests-coverage                    build DynaFlow Launcher and run coverage
+
+        =========== Tests
+        tests                                   launch DynaFlow Launcher unit tests
+
+        =========== Launch
+        launch-dir [dir]                        launch DynaFlow Launcher with:
+                                                - network filepath = <dir>/network.xml (only IIDM is supported)
+                                                - configuration filepath = <dir>/config.json
+        launch [network] [config]               launch DynaFlow Launcher
+
+        =========== Others
+        help                                    show all available options
+        version                                 show DynaFlow Launcher version
+        reset-environment                       reset all environment variables set by DynaFlow Launcher
+"
+}
+
+help() {
+    define_options
+    echo "$DYNAFLOW_LAUNCHER_USAGE"
+}
+
+set_commit_hook() {
+    $HERE/set_commit_hook.sh
+}
+
+set_environment() {
+    # dynawo vars
+    export DYNAWO_INSTALL_DIR=$DYNAWO_HOME
+    export LD_LIBRARY_PATH=$DYNAWO_INSTALL_DIR/lib:$LD_LIBRARY_PATH
+    export IIDM_XML_XSD_PATH=$DYNAWO_INSTALL_DIR/share/iidm/xsd
+    export DYNAWO_RESOURCES_DIR=$DYNAWO_INSTALL_DIR/share:$DYNAWO_INSTALL_DIR/share/xsd
+    export DYNAWO_DDB_DIR=$DYNAWO_INSTALL_DIR/ddb
+    export DYNAWO_IIDM_EXTENSION=$DYNAWO_INSTALL_DIR/lib/libdynawo_DataInterfaceIIDMExtension.so
+    export DYNAWO_LIBIIDM_EXTENSIONS=$DYNAWO_INSTALL_DIR/lib
+
+    # build
+    export DYNAFLOW_LAUNCHER_BUILD_DIR=$DYNAFLOW_LAUNCHER_HOME/buildLinux
+    export DYNAFLOW_LAUNCHER_INSTALL_DIR=$DYNAFLOW_LAUNCHER_HOME/installLinux
+    export_var_env DYNAFLOW_LAUNCHER_SHARED_LIB=OFF # same default value as cmakelist
+    export_var_env DYNAFLOW_LAUNCHER_USE_DOXYGEN=ON # same default value as cmakelist
+    export_var_env DYNAFLOW_LAUNCHER_BUILD_TESTS=ON # same default value as cmakelist
+    export_var_env DYNAFLOW_LAUNCHER_CMAKE_GENERATOR="Unix Makefiles"
+    export_var_env DYNAFLOW_LAUNCHER_PROCESSORS_USED=1
+
+    # Run
+    export DYNAFLOW_LAUNCHER_INSTALL=$DYNAFLOW_LAUNCHER_INSTALL_DIR
+    export LD_LIBRARY_PATH=$DYNAFLOW_LAUNCHER_HOME/lib64:$LD_LIBRARY_PATH
```

Comment:
better use a function like in dynawo

Reply:
and I think this append should be grouped together with the one with dfl lib, then it is better to see if the list that append and remove are the same

Reply:
I wanted to separate which is used for build and which is used for runtime.

Reply:
I prefer all LD_LIBRARY_PATH grouped together and add a comment to specify why it is used if you want. The first one will be used for build and runtime.

Reply:
I prefer all LD_LIBRARY_PATH grouped together and add a comment to specify why it is used if you want. The first one will be used for build and runtime.

Reply:
OK